Task 1.1

In [ ]:
import math
Mean_population = 100
SD_population = 15
n = 36 
x = 108
s = SD_population/(math.sqrt(n))
#H0 mean = 100  there is no effect of corn strach
#H1 mean > 100 there is effect of corn starch

Z = (x-Mean_population)/s

In [11]:
s,Z

(2.5, 3.2)

In [13]:
P_z= 0.9993 #from Ztable
p_lessthan_108 = 1-P_z
p_lessthan_108

0.0007000000000000339

p_lessthan_108 is less that alpha (0.05) we reject H0

Task 1.2

In [21]:
#Since we are interested in p(R) we denote them as P and P(D) as q
P1 = 0.52
q1 = 0.48
P2 = 0.47
q2 = 0.53
N = 100
#SD = Sqrt of pq
SD = math.sqrt((P1*q1)+(P2*q2))
#We need to find probability of P2 > P1
#P1-P2 < 0
x = 0
#Mean of differance of population is P1-P2
X = P1-P2
Z_P2_P1 = (x-X)/SD

In [28]:
print(f"Standard Deviation:{SD}, Z score :{Z_P2_P1}")

Standard Deviation:0.7061869440877535, Z score :-0.07080278164104213


P(Z<=x) is 0.24

Task 1.3

In [32]:
x = 1100
Mean_Popu = 1026
SD_popu = 209

Z = (x-Mean_Popu)/SD_popu
Z

0.35406698564593303

P(Z<=1100) is 0.638 or 63.8% </br>
you have better score than 63.8 % people who undertook the test

Task 2.1

In [45]:
from scipy.stats import chisquare,chi2_contingency
import pandas as pd

In [41]:
Data_observed = {"School":[60,40],"Bachelors":[54,44],"Masters":[46,53],"Ph.d.":[41,57]}
index_data=["Female","Male"]
df1 = pd.DataFrame(Data_observed,index =index_data )

In [55]:
'''H0 : The gender and education level are independent'''
'''H1 : The gender and education level are dependent'''
chi2_stat, p_val, dof, ex = chi2_contingency(df1)

In [60]:
print(f' expected value: {ex}')

Stat is :8.006066246262538, probability : 0.045886500891747214
dof : 3, expected value: [[50.88607595 49.86835443 50.37721519 49.86835443]
 [49.11392405 48.13164557 48.62278481 48.13164557]]


In [62]:
print(f'Stat is :{chi2_stat}, probability : {p_val},dof : {dof}')

Stat is :8.006066246262538, probability : 0.045886500891747214,dof : 3


degree of freedom = 3 </br>
Alpha = 0.05 </br>
Critical Value at degree of freedom = 7.818 (from chi sq table)</br>
so as 8.006066246262538 > 7.818 we reject H0 and conclude he gender and education level are dependent.


Task2.3

In [87]:
#Method 1
from scipy import stats
a = [10, 20, 30, 40, 50]
b = [5,10,15, 20, 25]
stats.f_oneway(a,b)

F_onewayResult(statistic=3.6, pvalue=0.0943497728424377)

In [74]:
#Method2
import statistics as stat
var_a = stat.variance(a)
var_b = stat.variance(b)
F = var_a/var_b
F

4.0

In [75]:
dof1 = len(a)-1
dof2 = len(b)-1

In [90]:
alpha = 0.05 #Or whatever you want your alpha to be.
p_value = stats.f.cdf(F, dof1, dof2)
if p_value > alpha:
     print(f'F = {F} , P = {p_value}, alpha  = {alpha}  Reject the null hypothesis Var(X) == Var(Y)' )
#that 

F = 4.0 , P = 0.896, alpha  = 0.05  Reject the null hypothesis Var(X) == Var(Y)


Task 2.2

In [91]:
Group1 = [51, 45, 33, 45, 67]
Group2 =  [23, 43, 23, 43, 45]
Group3 =  [56, 76, 74, 87, 56]

In [193]:
#Method1
F,p = stats.f_oneway(Group1,Group2,Group3)
F,p

(9.747205503009463, 0.0030597541434430556)

APA :- 
FCritical(2,12) = 7.95; 
P<0.05;

In [194]:
#Method2
import statistics
def SS_calc(k):
    ss = []
    mean = stat.mean(k)
    for x in k:
        dev= x - mean
        sq_dev = dev ** 2
        print(x,mean,dev,sq_dev)
        ss.append(sq_dev)
    SS = sum(ss)
    Var=SS/(len(k) - 1)
    
    return mean,SS,Var

In [186]:
SS_calc(Group1)

51 48.2 2.799999999999997 7.839999999999984
45 48.2 -3.200000000000003 10.240000000000018
33 48.2 -15.200000000000003 231.04000000000008
45 48.2 -3.200000000000003 10.240000000000018
67 48.2 18.799999999999997 353.4399999999999


(48.2, 612.8, 153.2)

In [187]:
SS_calc(Group2)

23 35.4 -12.399999999999999 153.75999999999996
43 35.4 7.600000000000001 57.76000000000002
23 35.4 -12.399999999999999 153.75999999999996
43 35.4 7.600000000000001 57.76000000000002
45 35.4 9.600000000000001 92.16000000000003


(35.4, 515.2, 128.8)

In [188]:
SS_calc(Group3)

56 69.8 -13.799999999999997 190.4399999999999
76 69.8 6.200000000000003 38.44000000000003
74 69.8 4.200000000000003 17.640000000000025
87 69.8 17.200000000000003 295.8400000000001
56 69.8 -13.799999999999997 190.4399999999999


(69.8, 732.8, 183.2)

In [200]:
MS_error = (153.2+128.8+183.2)/3
MS_error

155.06666666666666

In [191]:
SS_error = MS * 12
SS_error

1860.8

In [196]:
mean_ = [48.2,35.4,69.8]
SS_calc(mean_)

48.2 51.13333333333333 -2.93333333333333 8.604444444444425
35.4 51.13333333333333 -15.733333333333334 247.53777777777782
69.8 51.13333333333333 18.666666666666664 348.44444444444434


(51.13333333333333, 604.5866666666666, 302.2933333333333)

In [199]:
df_between = len(mean_)-1
MS_between = 302.3*5
SS_between = MS_between*df_between
df_between ,MS_between,SS_between

(2, 1511.5, 3023.0)

In [203]:
F = MS_between/MS_error
n_sq = SS_between/(SS_between+SS_error)
print(f'APA :- F={F}; Fcritical(2,12)=7.19 ; P<0.05; N square = {n_sq}')

APA :- F=9.747420464316424; Fcritical(2,12)=7.19 ; P<0.05; N square = 0.6189852164298292
